osu.Game/Overlays/Mods/ModSelectPanel.cs

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Audio;
+using osu.Framework.Audio.Sample;
+using osu.Framework.Bindables;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Colour;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Input.Events;
+using osu.Framework.Localisation;
+using osu.Framework.Utils;
+using osu.Game.Audio;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Graphics.UserInterface;
+using osuTK;
+using osuTK.Graphics;
+using osuTK.Input;
+
+namespace osu.Game.Overlays.Mods
+{
+    public abstract class ModSelectPanel : OsuClickableContainer, IHasAccentColour
+    {
+        public abstract BindableBool Active { get; }
+
+        public Color4 AccentColour { get; set; }
+
+        public LocalisableString Title
+        {
+            get => titleText.Text;
+            set => titleText.Text = value;
+        }
+
+        public LocalisableString Description
+        {
+            get => descriptionText.Text;
+            set => descriptionText.Text = value;
+        }
+
+        public const float CORNER_RADIUS = 7;
+
+        protected const float HEIGHT = 42;
+
+        protected virtual float IdleSwitchWidth => 14;
+        protected virtual float ExpandedSwitchWidth => 30;
```

Comment:
Would've made this abstract and overriden per implementation, but not a big deal.